packages/compass-connections/src/stores/connections-store.tsx

```diff
@@ -454,137 +696,91 @@ export function useConnections({
-      const findConnectionByFavoriteName = (name: string) =>
-        [...favoriteConnections, ...recentConnections].find(
-          (connection: ConnectionInfo) => connection.favorite?.name === name
-        );
-      const duplicate: ConnectionInfo = {
-        ...cloneDeep(connectionInfo),
-        id: new UUID().toString(),
-      };
+    [
+      connectionsManager,
+      disconnect,
+      editConnection,
+      getConnectionInfoById,
+      oidcAttemptConnectNotifyDeviceAuth,
+      oidcUpdateSecrets,
+      onConnectedRef,
+      onConnectionAttemptStartedRef,
+      onConnectionFailedRef,
+      openConnectionFailedToast,
+      openConnectionStartedToast,
+      openConnectionSucceededToast,
+      openMaximumConnectionsReachedToast,
+      preferences,
+      saveConnectionInfo,
+      showNonGenuineMongoDBWarningModal,
+    ]
+  );
 
-      if (duplicate.favorite?.name) {
-        const copyFormat = duplicate.favorite?.name.match(/(.*)\s\(([0-9])+\)/); // title (2) -> [title (2), title, 2]
-        const name = copyFormat ? copyFormat[1] : duplicate.favorite?.name;
-        let copyNumber = copyFormat ? parseInt(copyFormat[2]) : 1;
-        while (findConnectionByFavoriteName(`${name} (${copyNumber})`)) {
-          copyNumber++;
-        }
-        duplicate.favorite.name = `${name} (${copyNumber})`;
-      }
+  const connectRef = useCurrentRef(connect);
 
-      void saveConnectionInfo(duplicate).then(
-        () => {
-          dispatch({
-            type: 'set-active-connection',
-            connectionInfo: duplicate,
-          });
-        },
-        () => {
-          // We do nothing when if it fails
-        }
+  useEffect(() => {
+    if (connectionStorage.getAutoConnectInfo) {
+      void connectRef.current(
+        connectionStorage.getAutoConnectInfo.bind(connectionStorage)
       );
-    },
-    createDuplicateConnection(connectionInfo: ConnectionInfo) {
-      const findConnectionByFavoriteName = (name: string) =>
-        [...favoriteConnections, ...recentConnections].find(
-          (connection: ConnectionInfo) => connection.favorite?.name === name
-        );
-
-      const duplicate: ConnectionInfo = {
-        ...cloneDeep(connectionInfo),
-        id: new UUID().toString(),
-      };
+    }
+  }, [connectRef, connectionStorage]);
 
-      if (duplicate.favorite?.name) {
-        const copyFormat = duplicate.favorite?.name.match(/(.*)\s\(([0-9])+\)/); // title (2) -> [title (2), title, 2]
-        const name = copyFormat ? copyFormat[1] : duplicate.favorite?.name;
-        let copyNumber = copyFormat ? parseInt(copyFormat[2]) : 1;
-        while (findConnectionByFavoriteName(`${name} (${copyNumber})`)) {
-          copyNumber++;
-        }
-        duplicate.favorite.name = `${name} (${copyNumber})`;
+  const connectWithInflightCheck = useCallback(
+    async (connectionInfo: ConnectionInfo) => {
+      const inflightConnect = InFlightConnections.get(connectionInfo.id);
```

Comment:
@himanshusinghs is this something that you had in mind?

Reply:
yea something similar. I was probably thinking of having that in ConnectionsManager itself but having it here must have its own reasons. Sorry, I am yet to go through the PR (for my own understanding of the change).

Reply:
My thinking is that this connect method is the main and only interface that other parts of UI should use to start a connection, so debounce should happen as high as possible. We are also planning to dissolve manager and repository in the connections redux store with the follow-up refactor, so this logic will end up in the store eventually anyway